test/src/tests-audience-manager.ts

```diff
@@ -0,0 +1,172 @@
+import sinon from 'sinon';
+import fetchMock from 'fetch-mock/esm/client';
+import { expect } from 'chai';
+import { urls, apiKey, MPConfig, testMPID } from './config/constants';
+import Constants from '../../src/constants';
+import { MParticleWebSDK, SDKLoggerApi } from '../../src/sdkRuntimeModels';
+import AudienceManager, {
+    IAudienceMemberships, IAudienceMembershipsServerResponse
+} from '../../src/audienceManager';
+import Logger from '../../src/logger.js';
+
+declare global {
+    interface Window {
+        mParticle: MParticleWebSDK;
+        fetchMock: any;
+    }
+}
+
+const userAudienceUrl = `https://${Constants.DefaultBaseUrls.userAudienceUrl}${apiKey}/audience`;
+
+describe('AudienceManager', () => {
+    let mockServer;
+    before(function() {
+        fetchMock.restore();
+        sinon.restore();
+    });
+
+    beforeEach(function() {
+        fetchMock.restore();
+        mockServer = sinon.createFakeServer();
+        mockServer.respondImmediately = true;
+
+        mockServer.respondWith(urls.identify, [
+            200,
+            {},
+            JSON.stringify({ mpid: testMPID, is_logged_in: false }),
+        ]);
+
+        window.mParticle.config.flags = {
+            eventBatchingIntervalMillis: 1000,
+        };
+    });
+
+    afterEach(() => {
+        sinon.restore();
+        mockServer.reset();
+        fetchMock.restore();
+    });
+
+    it('should have proper properties on AudienceManager initialization', () => {
+        const newLogger: SDKLoggerApi = new Logger(window.mParticle.config);
+        const audienceManager = new AudienceManager(
+            Constants.DefaultBaseUrls.userAudienceUrl,
+            apiKey,
+            newLogger,
+        );
+
+        expect(audienceManager.logger).to.be.ok;
+        expect(audienceManager.url).to.equal(userAudienceUrl);
+        expect(audienceManager.userAudienceAPI).to.be.ok;
+    });
```

Comment:
This is a personal style choice, but I found that wrapping init logic in its own describe block makes it easier to collapse things in the IDE, and gives us room if we ever need to add new init tests in the future.

```suggestion
    describe('initialization', () => {
        it('should have proper properties on AudienceManager', () => {
            const newLogger: SDKLoggerApi = new Logger(window.mParticle.config);
            const audienceManager = new AudienceManager(
                Constants.DefaultBaseUrls.userAudienceUrl,
                apiKey,
                newLogger,
            );

            expect(audienceManager.logger).to.be.ok;
            expect(audienceManager.url).to.equal(userAudienceUrl);
            expect(audienceManager.userAudienceAPI).to.be.ok;
        });
    });
```